### PR TITLE
Throw PNSE in ThreadPoolBoundHandle.BindHandle on Unixes

### PR DESCRIPTION
**Pull request by kouvel**
Functional fix for https://github.com/dotnet/corefx/issues/19785

### CODE REVIEW

**Comment by kouvel** [comment]
Types in the S.T.Overlapped Unix assembly now type-forward to S.P.CoreLib, so this cleans up the one issue I found, to throw PNSE.

@alexperovich @stephentoub

File: src/mscorlib/src/System/Threading/ClrThreadPoolBoundHandle.cs [hunk 1]
```diff
@@ -77,34 +77,7 @@ public SafeHandle Handle
...
-                throw new ArgumentNullException(nameof(handle));
-
-            if (handle.IsClosed || handle.IsInvalid)
-                throw new ArgumentException(SR.Argument_InvalidHandle, nameof(handle));
-
-            try
-            {
-                // ThreadPool.BindHandle will always return true, otherwise, it throws. See the underlying FCall
-                // implementation in ThreadPoolNative::CorBindIoCompletionCallback to see the implementation.
-                bool succeeded = ThreadPool.BindHandle(handle);
```
Comment (stephentoub) [inline src/mscorlib/src/System/Threading/ClrThreadPoolBoundHandle.cs]: Should it actually be ThreadPool.BindHandle that throws the PNSE, with the rest of this method not changing?  Seems like we'd want the arg validation on handle, for example, to be consistent across platforms, even though then once you get past that on Unix you'd hit the PNSE.

Comment (kouvel) [inline src/mscorlib/src/System/Threading/ClrThreadPoolBoundHandle.cs]: ThreadPool.BindHandle returns true for success and false for failure, so on Unix it returns false. I'll move the PNSE check after the argument validation.

Comment (kouvel) [inline src/mscorlib/src/System/Threading/ClrThreadPoolBoundHandle.cs]: I'm fine with making ThreadPool.BindHandle throw PNSE though if that's what we want

Comment (stephentoub) [inline src/mscorlib/src/System/Threading/ClrThreadPoolBoundHandle.cs]: > ThreadPool.BindHandle returns true for success and false for failure, so on Unix it returns false. I'll move the PNSE check after the argument validation.

Ok

**Comment by kouvel** [comment]
@dotnet test Tizen armel Cross Debug Build

**Pull request merged**